stdlib/number.gr

```diff
@@ -439,3 +439,63 @@ export let sin = (radians: Number) => {
 export let cos = (radians: Number) => {
   sin(pi / 2 + radians)
 }
+
+/**
+ * Computes the tangent of a number (in radians) using Chebyshev polynomials.
+ *
+ * @param radians: The input in radians
+ * @returns The computed tangent
+ *
+ * @since v0.5.3
+ */
+export let tan = (radians: Number) => {
+  sin(radians) / cos(radians)
+}
+
+/**
+ * Computes the factorial of a number.
+ *
+ * @param input: The number to compute the factorial of, Must be a Non negative integer
+ * @returns The computed factorial
+ *
+ * @since v0.5.3
+ */
+export let factorial = input => {
+  // Error If It is Negative, or Not an Integer
+  if (sign(input) is -1 || isInteger(input)) {
+    fail "Factorial Expects A Non Negative Integer Input"
+  } else {
+    // Factorial Helper
+    let rec fact = n => if (n is 0) 1 else n * fact(n - 1)
+    // Perform Factorial
+    fact(input)
+  }
+}
+
+/**
+ * Clamps a number to a given range.
+ *
+ * @param min: The minimum value in the range
+ * @param max: The maximum value in the range
+ * @param input: The number to clamp
+ * @returns The clamped number
+ *
+ * @since v0.5.3
+ */
+export let clamp = (min: Number, max: Number, input: Number) => {
+  if (input < min) min else if (input > max) max else input
+}
+
+/**
+ * Linearly interpolates between the starting and ending value by the given amount.
+ *
+ * @param start: The starting value
+ * @param end: The ending value
+ * @param amount: an amount between 0 and 1, anything outside the range will produce an extrapolation i.e `lerp(0, 100, 2)` will produce 200
```

Comment:
this is a "interpolation" but you are saying that it also "exptrapolates". I don't like that.